### PR TITLE
Changing check mode behavior for some tasks which don't support it.

### PR DESCRIPTION
**Pull request by itayporezky**
##### SUMMARY
Fixing [#76127](https://github.com/ansible/ansible/issues/76127#issue-1034868052)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
copy
raw
fetch
reboot
script
wait_for_connection
win_copy

##### ADDITIONAL INFORMATION
[#76127](https://github.com/ansible/ansible/issues/76127#issue-1034868052)

Sorry for mistakes, first pull request.

### CODE REVIEW

File: lib/ansible/plugins/action/copy.py [hunk 1]
```diff
@@ -290,8 +290,7 @@ def _copy_file(self, source_full, source_rel, content, content_tempfile,
 
             if self._play_context.check_mode:
                 self._remove_tempfile_if_content_defined(content, content_tempfile)
-                result['changed'] = True
-                return result
+                raise AnsibleActionSkip('Check mode is not supported for this task.')
```
Comment (bcoca) [inline lib/ansible/plugins/action/copy.py:293]: This is incorrect, it should be
```
raise AnsibleActionDone(changed=True)
```

File: lib/ansible/plugins/action/raw.py [hunk 2]
```diff
@@ -34,9 +34,8 @@ def run(self, tmp=None, task_vars=None):
         del tmp  # tmp no longer has any effect
 
         if self._play_context.check_mode:
-            # in --check mode, always skip this module execution
-            result['skipped'] = True
-            return result
+            raise AnsibleActionSkip('Check mode is not supported for this task.')
+
```
Comment (bcoca) [inline lib/ansible/plugins/action/raw.py:38]: this is not needed as the 'super' already would trigger this from the base class, this is actually dead code and should be removed

File: lib/ansible/plugins/action/reboot.py [hunk 3]
```diff
@@ -412,7 +412,7 @@ def run(self, tmp=None, task_vars=None):
             return {'changed': False, 'elapsed': 0, 'rebooted': False, 'failed': True, 'msg': msg}
 
         if self._play_context.check_mode:
-            return {'changed': True, 'elapsed': 0, 'rebooted': True}
+            raise AnsibleActionSkip('Check mode is not supported for this task.')
```
Comment (bcoca) [inline lib/ansible/plugins/action/reboot.py:415]: not skip

```
raise AnsibleActionDone(changed=True, elapsed=0, rebooted=True)
```

File: lib/ansible/plugins/action/script.py [hunk 4]
```diff
@@ -119,7 +119,7 @@ def run(self, tmp=None, task_vars=None):
                     script_cmd = ' '.join([env_string, target_command])
 
             if self._play_context.check_mode:
-                raise _AnsibleActionDone()
+                raise AnsibleActionSkip('Check mode is not supported for this task.')
```
Comment (bcoca) [inline lib/ansible/plugins/action/script.py:122]:
```
raise AnsibleActionDone(changed=True)
```

File: lib/ansible/plugins/action/wait_for_connection.py [hunk 5]
```diff
@@ -70,8 +70,8 @@ def run(self, tmp=None, task_vars=None):
         timeout = int(self._task.args.get('timeout', self.DEFAULT_TIMEOUT))
 
         if self._play_context.check_mode:
-            display.vvv("wait_for_connection: skipping for check_mode")
-            return dict(skipped=True)
+            raise AnsibleActionSkip('Check mode is not supported for this task.')
+
```
Comment (bcoca) [inline lib/ansible/plugins/action/wait_for_connection.py:74]: can be removed, the call to 'supper' below already takes care of this

File: test/support/windows-integration/plugins/action/win_copy.py [hunk 6]
```diff
@@ -262,8 +262,7 @@ def _remove_tempfile_if_content_defined(self, content, content_tempfile):
 
     def _copy_single_file(self, local_file, dest, source_rel, task_vars, tmp, backup):
         if self._play_context.check_mode:
-            module_return = dict(changed=True)
-            return module_return
+            raise AnsibleActionSkip('Check mode is not supported for this task.')
```
Comment (bcoca) [inline test/support/windows-integration/plugins/action/win_copy.py:265]: these 2 are also wrong as it bypasses the logic on the caller that still does things with the resulting dictionary

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/action/copy.py (hunk 1 above)
Comment (s-hertel) [inline lib/ansible/plugins/action/copy.py:293]: Isn't this just removing check mode support? The only time check mode skipping makes sense to me (sorta; I think 'skipped' is overloaded) is if changed is unknown (such as for `script`), and here we know changed=True.

File: lib/ansible/plugins/action/reboot.py (hunk 3 above)
Comment (s-hertel) [inline lib/ansible/plugins/action/reboot.py:415]: We know reboot will always make a change so there's no need to disable check mode support. It's accurately reporting a change would occur. You can even see the intention a few lines above: `self._supports_check_mode = True`.

**Comment by mattclay** [comment]
Closing due to inactivity.

**Pull request closed without merging**